src/event_store/in_memory_event_store.rs

```diff
@@ -69,7 +72,7 @@ impl InMemoryEventStore {
         }
     }
 
-    fn event_batch(&mut self, size: usize) -> Result<EventBatch, Error> {
+    fn event_batch(&mut self, size: usize, retry_policy: RetryPolicy) -> Result<EventBatch, Error> {
```

Comment:
Nit: it's a bit unexpected for me that we need to pass the `retry_policy` here – doesn't sound relevant to what the function is doing. I understand that `EventBatch` takes it as an argument in the constructor, but wouldn't it be better to instead pass it as an argument to the `has_retry()` function?

Reply:
Agreed, that is much tidier.